core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -548,6 +554,9 @@ private[spark] class Executor(
           taskDescription.artifacts.jars,
           taskDescription.artifacts.archives,
           isolatedSession)
+        // Always reset the thread class loader to ensure if any updates, all threads (not only
+        // the thread that updated the dependencies) can update to the new class loader.
+        Thread.currentThread.setContextClassLoader(isolatedSession.replClassLoader)
```

Comment:
I am pretty sure we do this else where as well.

Are you also unsetting it once you are done?